### ISSUE TITLE
Missing magic methods

### ISSUE THREAD

**Issue by ariscn** [underyx/flask-redis#23]
With a real Redis instance, I can do `redis_store['foo']`. With a FlaskRedis instance, this fails.

FlaskRedis wraps Redis by doing:

```
    def __getattr__(self, name):
        return getattr(self._redis_client, name)
```

This works for all regular methods, but fails to account for magic methods. According to https://github.com/andymccurdy/redis-py/blob/master/redis/client.py, the supported methods are: `__repr__`, `__delitem__`, `__getitem__`, and `__setitem__`.

**Comment by underyx** [comment]
Thanks for the detailed feedback! Do you think we'd be better off using
__getattribute__ and writing code to fix normal operation on the class, or
should we just explicitly define these magic methods to redirect to the
Redis class (risking lack of support for any new magic methods the redis-py
project will implement later on)?

**Comment by ariscn** [comment]
The latter is the quickest fix. I literally just finished typing the following:

```
    def __repr__(self):
        return self._redis_client.__repr__()

    def __getattr__(self, name):
        return getattr(self._redis_client, name)

    def __setitem__(self, name, value):
        return self._redis_client.__setitem__(name, value)

    def __getitem__(self, name):
        return self._redis_client.__getitem__(name)

    def __delitem__(self, name):
        return self._redis_client.__delitem__(name)
```

I agree that risking breakage (due to changing function signature or new methods) isn't ideal. It looks like the Right Way to do it is a proxy metaclass? http://stackoverflow.com/questions/9057669/how-can-i-intercept-calls-to-pythons-magic-methods-in-new-style-classes

**Comment by underyx** [comment]
Hey @ariscn! Very, very late (I'm so sorry D:), but I'm committed a fix to this. I ended up pretty much just using your code from here, so I've set you as the author of the commit. See here: https://github.com/underyx/flask-redis/commit/92e417de4b7f2a06cad78da0691f0d60e02ea195

I did remove the `__repr__` definition though, since it felt wrong to *entirely* pretend that we're actually the `redis.StrictRedis` instance.
